mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -516,7 +532,7 @@ struct LockedAmountInfo {
 }
 
 struct SwapsContext {
-    running_swaps: Mutex<Vec<Weak<dyn AtomicSwap>>>,
+    running_swaps: Mutex<Vec<(Weak<dyn AtomicSwap>, AbortOnDropHandle)>>,
```

Comment:
Could you tell why do we need this change?

Is it related to some review note? may be I missed smth

Reply:
nah actually omar just told me we need the swaps to be stoppable via rpc (since we now do a run forever recovery), that's why we record their abort handles to be able to stop them mid-recover (or even mid-swap).